spr/src/commands/diff.rs

```diff
@@ -41,12 +42,54 @@ pub struct DiffOptions {
     #[clap(long, short = 'm')]
     message: Option<String>,
 
+    /// Which commits in the branch should be created/updated. This can be a
+    /// revspec such as HEAD~4..HEAD~1 or just one commit like HEAD~7.
+    #[clap(long, short = 'r')]
+    refs: Option<String>,
+
     /// Submit this commit as if it was cherry-picked on master. Do not base it
     /// on any intermediate changes between the master branch and this commit.
     #[clap(long)]
     cherry_pick: bool,
 }
 
+fn get_oids(refs: &String, repo: &git2::Repository) -> Result<HashSet<Oid>> {
+    // refs might be a single (eg 012345abc or HEAD) or a range (HEAD~4..HEAD~2)
+    let revspec = repo.revparse(&refs)?;
+    if revspec.mode().contains(git2::RevparseMode::MERGE_BASE) {
+        // this is ok, as other code doesn't handle merges
+        return Err(Error::new("Unable to cope with merge_base --refs"));
+    } else if revspec.mode().contains(git2::RevparseMode::SINGLE) {
+        // simple case, just return the id
+        return Ok(HashSet::from([revspec.from().unwrap().id()]));
```

Comment:
```suggestion
        return Ok(HashSet::from([revspec.from()?.id()]));
```
`unwrap()` is a bit frowned upon in production code. Given this function is returning a `Result` anyway, using `?` to pass on the error to the caller should work.

Reply:
will figure it out - but note from() is an option, not a result